mv test_git to 'destructive' integration tests

alikins opened the pull request:
##### ISSUE TYPE

<!--- Pick one below and delete the rest: -->
- Bugfix Pull Request
##### COMPONENT NAME

<!--- Name of the plugin/module/task -->

test/integration/
##### ANSIBLE VERSION

<!--- Paste verbatim output from “ansible --version” between quotes below -->

```
ansible 2.2.0 (test_git_to_destructive 752e62ac63) last updated 2016/09/12 14:35:45 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 1a0e15094f) last updated 2016/09/12 10:00:41 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD f83aa9fff3) last updated 2016/09/09 09:07:48 (GMT -400)
  config file = /home/adrian/.ansible.cfg
  configured module search path = ['/home/adrian/src/ansible-modules-core', '/home/adrian/src/ansible-modules-extras']


```
##### SUMMARY

The test_git role removes ~/.ssh/known_hosts currently
and that is destructive.

gundalow commented:
I've marked this as P1 as we shouldn't have tests that mess with `~/.ssh/`, as we've previously discussed on IRC.

alikins committed:
mv test_git to 'destructive' integration tests

The test_git role removes ~/.ssh/known_hosts currently
and that is destructive.

alikins commented:
I'll add a note that aside from mucking with known_hosts, it also modifies ~/.gitconfig, setting up a 'Ansible Test Runner' user. 

You may recognize that user name from such commits as 'mv test_git to 'destructive' integration tests'.
Grrr.

alikins commented:
Merging.
